packages/theme/src/cli/services/profile.test.ts

```diff
@@ -0,0 +1,107 @@
+import {profile} from './profile.js'
+import {render} from '../utilities/theme-environment/storefront-renderer.js'
+import {ensureAuthenticatedStorefront} from '@shopify/cli-kit/node/session'
+import {openURL} from '@shopify/cli-kit/node/system'
+import {vi, describe, expect, beforeEach, test} from 'vitest'
+import {Headers, Response} from 'node-fetch'
+import {readFile} from 'fs/promises'
+
+vi.mock('@shopify/cli-kit/node/session')
+vi.mock('@shopify/cli-kit/node/system')
+vi.mock('../utilities/theme-environment/storefront-password-prompt.js')
+vi.mock('../utilities/theme-environment/storefront-session.js')
+vi.mock('../utilities/theme-environment/storefront-renderer.js')
+
+describe('profile', () => {
+  const mockProfileData = {
+    name: 'test-profile',
+    data: 'sample-data',
+  }
+  const mockToken = 'mock-token'
+  const mockAdminSession = {token: mockToken, storeFqdn: 'test-store.myshopify.com'}
+  const themeId = '123'
+  const urlPath = '/products/test'
+
+  beforeEach(() => {
+    vi.mocked(ensureAuthenticatedStorefront).mockResolvedValue(mockToken)
+    vi.mocked(render).mockResolvedValue(
+      new Response(JSON.stringify(mockProfileData), {
+        status: 200,
+        headers: new Headers({
+          'content-type': 'application/json',
+        }),
+      }),
+    )
+  })
+
+  test('outputs JSON to stdout when asJson is true', async () => {
```

Comment:
In this repo the convention is to follow `//given //when //then`

I'm not fussed about this - feel free to ship and I can refactor that